ruby_event_store/lib/ruby_event_store/client.rb

```diff
@@ -198,13 +208,27 @@ def normalize_to_array(events)
     end
 
     def enrich_event_metadata(event)
-      event.metadata[:timestamp] ||= clock.()
       if metadata_proc
+        warn "`RubyEventStore::Client#metadata_proc` has been deprecated. Use `RubyEventStore::Client#with_metadata` instead."
```

Comment:
I am not sure we want to deprecate it... Let me think/check.

Reply:
hmm too compilicated for quick review:

* https://github.com/RailsEventStore/rails_event_store/blob/3b191017dd854607b88c5e01e9328a4f13789a98/rails_event_store/lib/rails_event_store/middleware.rb#L9
* https://github.com/RailsEventStore/rails_event_store/blob/3b191017dd854607b88c5e01e9328a4f13789a98/rails_event_store/lib/rails_event_store/client.rb#L7